prow/pubsub/reporter/reporter.go

```diff
@@ -124,21 +124,14 @@ func (c *Client) Report(pj *prowapi.ProwJob) ([]*prowapi.ProwJob, error) {
 func (c *Client) generateMessageFromPJ(pj *prowapi.ProwJob) *ReportMessage {
 	pubSubMap := findLabels(pj, PubSubProjectLabel, PubSubTopicLabel, PubSubRunIDLabel)
 
-	var number int
-	if (pj.Spec.Refs == nil) || len(pj.Spec.Refs.Pulls) == 0 {
-		number = 0
-	} else {
-		number = pj.Spec.Refs.Pulls[0].Number
-	}
-
 	return &ReportMessage{
-		Project:  pubSubMap[PubSubProjectLabel],
-		Topic:    pubSubMap[PubSubTopicLabel],
-		RunID:    pubSubMap[PubSubRunIDLabel],
-		Status:   pj.Status.State,
-		URL:      pj.Status.URL,
-		GCSPath:  strings.Replace(pj.Status.URL, c.config().Plank.GetJobURLPrefix(pj.Spec.Refs), GCSPrefix, 1),
-		PRNumber: number,
-		JobType:  pj.Spec.Type,
+		Project: pubSubMap[PubSubProjectLabel],
+		Topic:   pubSubMap[PubSubTopicLabel],
+		RunID:   pubSubMap[PubSubRunIDLabel],
+		Status:  pj.Status.State,
+		URL:     pj.Status.URL,
+		GCSPath: strings.Replace(pj.Status.URL, c.config().Plank.GetJobURLPrefix(pj.Spec.Refs), GCSPrefix, 1),
+		Refs:    pj.Spec.Refs,
```

Comment:
To be totally correct I think it would be nice to use `[]prowapi.Refs` and concatenate `pj.Spec.Refs` with `pj.Spec.ExtraRefs`

Reply:
if we combine them, will only the first ref contain the PR details and the rest of the refs will have empty Pull[]?

Reply:
Something like this will have only a slice of valid refs:

```go
var refs []prowapi.Refs
if pj.Spec.Refs != nil {
    refs = append(refs, pj.Spec.Refs)
}
refs = append(refs, pj.Spec.ExtraRefs...)
```

Reply:
thanks. updated